structs.go

```diff
@@ -1,8 +1,17 @@
 package prismgoclient
 
+type AuthType string
+
+const (
+	AuthTypeBasic  AuthType = "basic_auth"
+	AuthTypeAPIKey AuthType = "api_key"
+)
+
 // Credentials needed username and password
 type Credentials struct {
 	URL                string
+	AuthType           AuthType
```

Comment:
I don't think we need an explicit auth type here. If API Key is present, prioritize that, if not fallback to Username and password